R/last_fit.R

```diff
@@ -65,42 +69,66 @@ last_fit <- function(object, ...) {
 last_fit.default <- function(object, ...) {
   empty_ellipses(...)
   msg <- paste0(
-    "The first argument to [last_fit()] should be either a ",
-    "formula, recipe, or workflow."
+    "The first argument to [last_fit()] should be either ",
+    "a model or workflow."
   )
   rlang::abort(msg)
 }
 
 #' @rdname last_fit
 #' @export
 last_fit.recipe <- function(object, model, split, ..., metrics = NULL) {
+  lifecycle::deprecate_soft("0.1.0",
+                            what = "last_fit.recipe()",
+                            details = deprecate_msg(match.call(), "last_fit"))
```

Comment:
What is the call manipulation getting us?

Would it be any simpler to do this?

```r
deprecate_msg <- function(what) {
  glue::glue(
    "As of tune 0.1.0, the first argument to `{what}()` ",
    "should be either a parsnip model or a workflow."
  )
}
```

Then use 

```r
deprecate_soft("0.1.0", what = "last_fit.recipe()", with = "last_fit.model_spec()", details = deprecate_msg("last_fit"))
```

Reply:
The call manipulation lets us give the user the new call they should use, [shown up here](https://github.com/tidymodels/tune/pull/170#issuecomment-587735689). It's not a huge deal either way and it would be simpler to not mess with the call, but my opinion is that it's nice to show the user what to do (can copy/paste the result we give them).

Reply:
Oh now I get it. Yea that seems useful!